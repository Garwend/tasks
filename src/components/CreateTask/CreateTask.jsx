import React, { useState } from "react";

const CreateTask = ({ setWaitingTasks, setCurrentTasks }) => {
    const [priority, setPriority] = useState(1);

    const handleChange = (e) => {
        setPriority(Number(e.target.value));
    }

    const handleClick = () => {
        const seconds = Math.floor(Math.random() * 6) + 5;
        const task = { id: Date.now(), title: seconds, seconds, priority };

        setCurrentTasks(prevState => {
            if (prevState.length >= 10) {
                setWaitingTasks(prevWaitingTasks => {
                    if (prevWaitingTasks.length === 0) return [task]
                    let i = 0
                    for (i; i < prevWaitingTasks.length; i++) {
                        if (prevWaitingTasks[i].priority <= priority) {
                            break;
                        }
                    }
                    const waitingTasks = prevWaitingTasks.slice(0)
                    waitingTasks.splice(i, 0, task)

                    return waitingTasks
                })
                return prevState
            } else {
                return [...prevState, task]
            }
        })
    }

    return (
        <section>
            <input type="number" name="priority" id="priority" min="1" data-testid="priority" onChange={handleChange} value={priority} />
            <button onClick={handleClick}>dodaj zadanie</button>
        </section>
    )
}

export default CreateTask;