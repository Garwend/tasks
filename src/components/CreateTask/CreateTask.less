@add-color: green;
@add-color-hover: rgb(0, 104, 0);
@on-add-color: white;

.flex-center-column {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.create-task-wrap {
    height: 30vh;
    .flex-center-column();

    input {
        font-size: 40px;
        padding: 5px;
        text-align: center;
    }

    button {
        padding: 8px;
        border: none;
        margin-top: 30px;
        border-radius: 10px;
        font-size: 20px;
        background-color: @add-color;
        color: @on-add-color;
        transition: .3s;
        cursor: pointer;

        &:hover {
            background-color: @add-color-hover;
        }
    }
}