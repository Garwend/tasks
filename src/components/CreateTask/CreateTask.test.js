import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CreateTask from "./CreateTask";

afterEach(cleanup);

test('priority value', () => {
    render(<CreateTask />);
    const numberInput = screen.getByTestId('priority');
    fireEvent.change(numberInput, {target: {value: '23'}})
    expect(numberInput.value).toBe('23')
})

test('should render CreateTask component', () => {
    const tree = renderer.create(<CreateTask />).toJSON();
    expect(tree).toMatchSnapshot();
})