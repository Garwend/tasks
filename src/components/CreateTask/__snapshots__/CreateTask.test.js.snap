// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreateTask component 1`] = `
<section>
  <input
    data-testid="priority"
    id="priority"
    min="1"
    name="priority"
    onChange={[Function]}
    type="number"
    value={1}
  />
  <button
    onClick={[Function]}
  >
    dodaj zadanie
  </button>
</section>
`;
